ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -695,25 +707,73 @@ private Dataset executeGetDatasetStatement(PreparedStatement statement, int acco
         retentionCount, userTags);
   }
 
+  /**
+   * Get version schema corresponding to the dataset.
+   * @param statement the mysql statement to get the version schema of the dataset.
+   * @param accountId the id of the parent account.
+   * @param containerId the if of the container.
+   * @param datasetName the name of the dataset.
+   * @return the name of the dataset.
+   * @throws SQLException
+   */
+  private Dataset.VersionSchema executeGetVersionSchema(PreparedStatement statement, int accountId, int containerId,
+      String datasetName) throws SQLException {
+    ResultSet resultSet = null;
+    Dataset.VersionSchema versionSchema;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, datasetName);
+      resultSet = statement.executeQuery();
+      resultSet.next();
+      versionSchema = Dataset.VersionSchema.values()[resultSet.getInt(VERSION_SCHEMA)];
+    } finally {
+      //If result set is not created in a try-with-resources block, it needs to be closed in a finally block.
+      closeQuietly(resultSet);
+    }
+    return versionSchema;
+  }
+
+  /**
+   * Convert string version to long value.
+   * @param version The string version.
+   * @param versionSchema The {@link com.github.ambry.account.Dataset.VersionSchema} of the dataset.
+   * @return the long value of version.
+   */
+  private long getVersionBasedOnSchema(String version, Dataset.VersionSchema versionSchema) {
+    switch (versionSchema) {
+      case TIMESTAMP:
+      case MONOTONIC:
+        return Long.parseLong(version);
+      case SEMANTIC:
+        String[] versionArray = version.split("[.]", 3);
+        return Long.parseLong(versionArray[0]) * 1000000 + Long.parseLong(versionArray[1]) * 1000 + Integer.parseInt(
+            versionArray[2]);
```

Comment:
can you explain what the semantic version schema mean?

Reply:
Something like this: https://semver.org/
Basically the version is like X.Y.Z.
And I'm planning to support this MAJOR, MINOR, PATCH version increment in future rb.